lib/ui/text.dart

```diff
@@ -106,184 +105,1306 @@ class FontWeight {
+///             Text('but old-style figures blend well with lower case:',
+///                 style: titleStyle),
+///             Text('The 1972 cup final was a 1-1 draw.',
+///                 style: TextStyle(
+///                     fontFamily: 'Cardo',
+///                     fontSize: 24,
+///                     fontFeatures: [FontFeature.oldstyleFigures()])),
+///             Spacer(),
+///             Divider(),
+///             Spacer(),
+///             Text('fractions look better with a custom ligature:',
+///                 style: titleStyle),
+///             Text('Add 1/2 tsp of flour and stir.',
+///                 style: TextStyle(
+///                     fontFamily: 'Milonga',
+///                     fontSize: 24,
+///                     fontFeatures: [FontFeature.alternativeFractions()])),
+///             Spacer(),
+///             Divider(),
+///             Spacer(),
+///             Text('multiple stylistic sets in one font:', style: titleStyle),
+///             Text('Raleway Dots',
+///                 style: TextStyle(fontFamily: 'Raleway Dots', fontSize: 48)),
+///             Text('Raleway Dots',
+///                 style: TextStyle(
+///                   fontFeatures: [FontFeature.stylisticSet(1)],
+///                   fontFamily: 'Raleway Dots',
+///                   fontSize: 48,
+///                 )),
+///             Spacer(flex: 5),
+///           ],
+///         ),
+///       ),
+///     );
+///   }
+/// }
 /// ```
 /// {@end-tool}
+///
+/// See also:
+///
+///  * <https://en.wikipedia.org/wiki/List_of_typographic_features>,
+///    Wikipedia's description of these typographic features.
+///
+///  * <https://docs.microsoft.com/en-us/typography/opentype/spec/featuretags>,
+///    Microsoft's registry of these features.
 class FontFeature {
   /// Creates a [FontFeature] object, which can be added to a [TextStyle] to
   /// change how the engine selects glyphs when rendering text.
   ///
   /// `feature` is the four-character tag that identifies the feature.
   /// These tags are specified by font formats such as OpenType.
   ///
-  /// `value` is the value that the feature will be set to.  The behavior
-  /// of the value depends on the specific feature.  Many features are
+  /// `value` is the value that the feature will be set to. The behavior
+  /// of the value depends on the specific feature. Many features are
   /// flags whose value can be 1 (when enabled) or 0 (when disabled).
   ///
   /// See <https://docs.microsoft.com/en-us/typography/opentype/spec/featuretags>
-  // ignore: unnecessary_null_comparison
-  const FontFeature(this.feature, [ this.value = 1 ]) : assert(feature != null), assert(feature.length == 4), assert(value != null), assert(value >= 0);
+  const FontFeature(
+    this.feature,
+    [ this.value = 1 ]
+  ) : assert(feature != null), // ignore: unnecessary_null_comparison
+      assert(feature.length == 4, 'Feature tag must be exactly four characters long.'),
```

Comment:
These are nice to add, but I'm pretty sure they just show up as `<optimized out>` to users when thrown from dart:ui. I'm not sure why and whether that's on us or on Dart, or what the cost of fixing it would be.

Reply:
Is it the messages or the expressions that are optimized out? Any idea?

Reply:
Just messages